spring/spring4/spring4-common/src/main/java/org/infinispan/spring/provider/CacheDelegate.java

```diff
@@ -71,13 +76,33 @@ public ValueWrapper get(final Object key) {
 
    @Override
    public <T> T get(Object key, Callable<T> valueLoader) {
-      return (T) nativeCache.computeIfAbsent(key, keyToBeInserted -> {
-         try {
-            return valueLoader.call();
-         } catch (Exception e) {
-            throw ValueRetrievalExceptionResolver.throwValueRetrievalException(key, valueLoader, e);
+      long stamp = synchronousGetLock.readLock();
+      try {
+         T value = (T) nativeCache.get(key);
+         if (value == null) {
+            AtomicBoolean valueObtained = new AtomicBoolean();
```

Comment:
Actually you shouldn't use `ByRef.Boolean`, just a `break`. If I am not mistaken the `valueObtained` does not escape the scope of the current looping thread.